spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/schema/CypherGenerator.java

```diff
@@ -328,27 +330,30 @@ public Expression createReturnStatementForMatch(NodeDescription<?> nodeDescripti
 
 		Set<RelationshipDescription> processedRelationships = new HashSet<>();
 
-		return projectPropertiesAndRelationships(nodeDescription, NAME_OF_ROOT_NODE, includeField, processedRelationships, 0);
+		return projectPropertiesAndRelationships(nodeDescription, NAME_OF_ROOT_NODE, includeField,
+			processedRelationships, new HashMap<>());
```

Comment:
Could thread safety be an issue?

```suggestion
			processedRelationships, new ConcurrentHashMap<>());
```